app/modulo.hoon

```diff
@@ -0,0 +1,212 @@
+/-  *modulo
+/+  *server
+/=  index
+  /^  octs
+  /;  as-octs:mimes:html
+  /:  /===/app/modulo/index
+  /|  /html/
+      /~  ~
+  ==
+/=  modulo-js
+  /^  octs
+  /;  as-octs:mimes:html
+  /:  /===/app/modulo/script
+  /|  /js/
+      /~  ~
+  ==
+=,  format
+|%
+:: +move: output effect
+::
++$  move  [bone card]
+:: +card: output effect payload
+::
++$  card
+  $%  [%connect wire binding:http-server term]
+      [%serve wire binding:http-server generator:http-server]
+      [%disconnect wire binding:http-server]
+      [%http-response =http-event:http]
+      [%poke wire dock poke]
+      [%diff %json json]
+  ==
+
++$  poke
+  $%  [%modulo-bind app=term]
+      [%modulo-unbind app=term]
+  ==
+::
++$  state
+  $%  $:  %0   
+          session=(map term @t)
+          order=(list term)
+          cur=(unit [term @])
+      ==
+  ==
+::
+++  session-as-json
+  |=  [cur=(unit [term @]) session=(map term @t) order=(list term)]
+  ^-  json
+  ?~  cur
+    *json
+  %-  pairs:enjs
+  :~  [%app %s -.u.cur]
+      [%url %s (~(got by session) -.u.cur)]
+      :-  %list
+        :-  %a
+      %+  turn  order
+        |=  [a=term]
+        [%s a]
+  ==
+::
+--
+::
+|_  [bow=bowl:gall sta=state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit *)
+  ^-  (quip move _this)
+  ?~  old
+    :_  this
+    :~  [ost.bow %connect / [~ /] %modulo]
+        [ost.bow %connect / [~ /'~landscape'] %landscape]
```

Comment:
Any particular reason this binding is created here instead of in `app/landscape`?

Reply:
No, just leftover from the brief detour into making landscape a generator. Went back to an app for the sake of cross-ship posting, I'll change this.